core/constraints/validation.go

```diff
@@ -195,7 +199,7 @@ func (v *validator) checkUnsupported(cons Value) []string {
 // checkValidValues returns an error if the constraints value contains an
 // attribute value which is not allowed by the vocab which may have been
 // registered for it.
-func (v *validator) checkValidValues(cons Value) error {
+func (v *validator) checkValidValues(cons Value) *ValidatorInvalidConstraintErr {
```

Comment:
No need for this change. It's just an error return. The caller can do `errors.As()` if needed.
But the comment should describe the error that is returned. eg here's an example from elsewhere
```
// CreateModel is responsible for creating a new model within the model
// database.
//
// The following error types can be expected to be returned:
// - [modelerrors.AlreadyExists]: When the model uuid is already in use.
func (s *ModelService) CreateModel(
```

You'll also need to add the extra comment to the `Validate()` interface method and implementation.